Core/src/Surfaces/CylinderSurface.cpp

```diff
@@ -111,9 +111,13 @@ Acts::Vector3D Acts::CylinderSurface::localToGlobal(
 
 Acts::Result<Acts::Vector2D> Acts::CylinderSurface::globalToLocal(
     const GeometryContext& gctx, const Vector3D& position,
-    const Vector3D& /*unused*/) const {
-  // @todo check if s_onSurfaceTolerance would do here
-  double inttol = bounds().get(CylinderBounds::eR) * 0.0001;
+    const Vector3D& /*unused*/, double tolerance) const {
+  double inttol = tolerance;
+  if (tolerance == s_onSurfaceTolerance) {
+    // transform default value!
+    // @TODO: check if s_onSurfaceTolerance would do here
+    inttol = bounds().get(CylinderBounds::eR) * 0.0001;
```

Comment:
That actually comes from ATLAS, I think we try in a separate PR if we omit this awkward scaling.

Reply:
Yeah, this just replicates the current behavior if no explicit tolerance is given, without going back and questioning if this here make sense anymore.